zebra-chain/src/notes/sapling/arbitrary.rs

```diff
@@ -0,0 +1,35 @@
+use proptest::{arbitrary::any, collection::vec, prelude::*};
+
+use crate::notes::sapling;
+
+impl Arbitrary for sapling::EncryptedCiphertext {
+    type Parameters = ();
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        (vec(any::<u8>(), 580))
+            .prop_map(|v| {
+                let mut bytes = [0; 580];
+                bytes.copy_from_slice(v.as_slice());
+                Self(bytes)
+            })
+            .boxed()
+    }
```

Comment:
Note that this will essentially always produce ciphertexts that don't decrypt with any key. Not sure how to get good test coverage for valid ciphertexts using property-based testing; perhaps that requires conventional testing.

Reply:
That's fine for now, the basic strategy is mostly for testing serialization roundtrips etc

Reply:
In the future we could use an enum strategy to generate a mix of valid and invalid ciphertexts.